packages/coin-sol/src/index.ts

```diff
@@ -0,0 +1,39 @@
+import { coin as COIN, error as ERROR, Transport } from '@coolwallet/core';
+import { signTransaction } from './sign';
+import * as types from './config/types';
+import * as params from './config/params';
+import * as stringUtil from './utils/stringUtil';
+import TransactionCreator from './utils/TransactionCreator';
+
+export { types, TransactionCreator };
+
+export default class SOL extends COIN.EDDSACoin implements COIN.Coin {
+  constructor() {
+    super(params.COIN_TYPE);
+  }
+
+  async getAddress(transport: Transport, appPrivateKey: string, appId: string): Promise<string> {
+    const publicKey = await this.getPublicKey(transport, appPrivateKey, appId, true);
+
+    if (!publicKey) {
+      throw new ERROR.SDKError(this.getAddress.name, 'public key is undefined');
+    }
+    return this.getAddressByAccountKey(publicKey);
+  }
+
+  async getAddressByAccountKey(publicKey: string): Promise<string> {
+    return stringUtil.pubKeyToAddress(publicKey);
+  }
+
+  async signTransaction(signTxData: types.signTxType): Promise<Buffer> {
+    const { transaction } = signTxData;
+    switch (transaction.txType) {
+      case params.TRANSACTION_TYPE.SPL_TOKEN:
+        return signTransaction(signTxData, params.TRANSACTION_TYPE.SPL_TOKEN);
```

Comment:
When signing the SPL token transfer, we would like to show the symbol of the token just like how erc20 doing.

ex.
| SOL | USDT | 111111 |           0 |
|-----|------|--------|-------------|
| SPL |      | ADDR   |    0.000001 |

Reply:
I see but do we need to add signature for each token, and if we do, how to get valid signature for each token
![image](https://user-images.githubusercontent.com/88299386/163359060-d2d2c298-7ade-408c-824e-35142cce909b.png)

Reply:
If there is no signature, It will show like this:

ex.
| SOL | @ USDT | 111111 |           0 |
|-----|------|--------|-------------|
| SPL |      | ADDR   |    0.000001 |

If the token has a "@" character means it is not an official support token by CoolWallet.

Reply:
Got it, I would update this logic asap

Reply:
Here is what the expected data looks like:

| Fields               | Bytes | Description                                               | Example                                                                                                          |
|----------------------|-------|-----------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
| TokenDecimal         | 1     | Unit decimal to hex                                       | 6 -> 06                                                                                                          |
| TokenNameLength      | 1     | Length of symbol                                          | USDT -> 04                                                                                                       |
| TokenName            | 7     | Symbol ASCII text to hex, adding 0 the end to fit 14 bits | USDT -> 55 53 44 54 -> 55534454000000                                                                            |
| TokenContractAddress | 32    | Token Public Key                                          | Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB -> ce010e60afedb22717bd63192f54145a3f965a33bb82d2c7029eb2ce1e208264 |

Reply:
@hank121314 I have recently updated the logic we discussed, please check it again